lib/utils/color.dart

```diff
@@ -8,14 +8,9 @@ import 'package:flutter/material.dart';
 /// Inspired by https://stackoverflow.com/a/13532993
 Color shadeColor(double factor, Color color, {int minBit = 0, int maxBit = 0xff}) {
   assert(factor >= -1 && factor <= 1);
-  var r = ((color.value >> 16) & 0xff).clamp(0, 0xff);
-  var g = ((color.value >> 8) & 0xff).clamp(0, 0xff);
-  var b = (color.value & 0xff).clamp(0, 0xff);
-
   factor = (1 + factor);
-  r = (r * factor).toInt().clamp(minBit, maxBit);
-  g = (g * factor).toInt().clamp(minBit, maxBit);
-  b = (b * factor).toInt().clamp(minBit, maxBit);
-
-  return Color((0xff << 24) + (r << 16) + (g << 8) + b);
+  final r = (color.r * 255.0 * factor).toInt().clamp(minBit, maxBit);
+  final g = (color.g * 255.0 * factor).toInt().clamp(minBit, maxBit);
+  final b = (color.b * 255.0 * factor).toInt().clamp(minBit, maxBit);
```

Comment:
Seems like we no logner need int conversion here

Reply:
`fromARGB` takes an `int`, and without the `toInt` these expressions produce a `num`.

But maybe it would be better if we convert the `minBit` and `maxBit` into `0.0`-`1.0` domain and just use floats everywhere. The only reason why I didn't do it is because we loose the `floor` operation performed by `toInt`, so the results will be slightly different from before.

Reply:
Are you sure we still need to use `fromARGB` and not `from`?

Reply:
`from` takes values in the `0.0` to `1.0` domain. We can use that if we avoid the multiplication by `255` and if we convert the `minBit` and `maxBit` to floats in that domain. I think it's the better solution, but it will change the output of this function slightly, since we won't have the `floor` from the `toInt` anymore.

Should we also move this into a separate issue?

Reply:
Or maybe the same issue about the alpha channel?

Reply:
Let's do these 2 refactors in the separate issue and leave it as-is in this PR